services/scribe/node/scribe_test.go

```diff
@@ -15,13 +19,114 @@ import (
 	"github.com/synapsecns/sanguine/services/scribe/testutil/testcontract"
 	"math/big"
 	"os"
+	"sync"
 	"time"
 )
 
+// TODO combine these functions with backfill/backend as well as other tests
+
+// ReachBlockHeight reaches a block height on a backend.
+func (l *LiveSuite) ReachBlockHeight(ctx context.Context, backend backends.SimulatedTestBackend, desiredBlockHeight uint64) {
+	i := 0
+	for {
+		select {
+		case <-ctx.Done():
+			l.T().Log(ctx.Err())
+			return
+		default:
+			// continue
+		}
+		i++
+		backend.FundAccount(ctx, common.BigToAddress(big.NewInt(int64(i))), *big.NewInt(params.Wei))
+
+		latestBlock, err := backend.BlockNumber(ctx)
+		Nil(l.T(), err)
+
+		if latestBlock >= desiredBlockHeight {
+			return
+		}
+	}
+}
+
+// startOmnirpcServer boots an omnirpc server for an rpc address.
+// the url for this rpc is returned.
+func (l *LiveSuite) startOmnirpcServer(ctx context.Context, backend backends.SimulatedTestBackend) string {
+	baseHost := testhelper.NewOmnirpcServer(ctx, l.T(), backend)
+	return testhelper.GetURL(baseHost, backend)
+}
+
+// ReachBlockHeight reaches a block height on a backend.
+func (l *LiveSuite) PopuluateWithLogs(ctx context.Context, backend backends.SimulatedTestBackend, desiredBlockHeight uint64) common.Address {
+	i := 0
+	var address common.Address
+	for {
+		select {
+		case <-ctx.Done():
+			l.T().Log(ctx.Err())
+			return address
+		default:
+			// continue
+		}
+		i++
+		backend.FundAccount(ctx, common.BigToAddress(big.NewInt(int64(i))), *big.NewInt(params.Wei))
+		testContract, testRef := l.manager.GetTestContract(l.GetTestContext(), backend)
+		address = testContract.Address()
+		transactOpts := backend.GetTxContext(l.GetTestContext(), nil)
+		tx, err := testRef.EmitEventA(transactOpts.TransactOpts, big.NewInt(1), big.NewInt(2), big.NewInt(3))
+		Nil(l.T(), err)
+		backend.WaitForConfirmation(l.GetTestContext(), tx)
+
+		latestBlock, err := backend.BlockNumber(ctx)
+		Nil(l.T(), err)
+
+		if latestBlock >= desiredBlockHeight {
+			return address
+		}
+	}
+}
+func (l *LiveSuite) TestGetBlockHashes() {
+	testBackend := geth.NewEmbeddedBackend(l.GetTestContext(), l.T())
```

Comment:
Tests 
- blockhash batching
- livefill for 20 sec and checks db for content
- livefill 20 sec and checks db for content, verifies confirmations, checks latest block stats
- livefill for 20 sec with a receipt with an invalid blockhash, verifies it gets removed
- adding a livefill test with a prod config